src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Withdraw/components/Confirm.tsx

```diff
@@ -77,101 +97,328 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
+
+      const { dust_amount } = quote
+
+      const accountAddress = await getAccountAddress()
+
+      const sendInput: SendInput = {
+        cryptoAmount: fromBaseUnit(dust_amount, asset.precision),
+        asset,
+        to: quote.inbound_address,
+        from: accountAddress,
+        sendMax: false,
+        accountId: accountId ?? '',
+        amountFieldError: '',
+        cryptoSymbol: asset?.symbol ?? '',
+        estimatedFees,
+        feeType: FeeDataKey.Fast,
+        fiatAmount: '',
+        fiatSymbol: selectedCurrency,
+        vanityAddress: '',
+        input: quote.inbound_address,
+      }
+
+      return sendInput
+    } catch (e) {
+      moduleLogger.error({ fn: 'getSendInput', e }, 'Error building THORChain savers Tx')
+    }
+  }, [
+    accountId,
+    assetId,
+    getEstimateFeesArgs,
+    state?.withdraw.cryptoAmount,
+    asset,
+    opportunityData?.stakedAmountCryptoBaseUnit,
+    opportunityData?.rewardsAmountsCryptoBaseUnit,
+    getAccountAddress,
+    selectedCurrency,
+  ])
+
+  const handleMultiTxSend = useCallback(async (): Promise<string | undefined> => {
+    if (!walletState.wallet) return
+
+    // THORChain Txs need to always be sent from the same address, since the address (NOT the pubkey) is used to identify an active position
+    // The way THORChain does this is by not being xpub-compliant, and only exposing a single address for UTXOs in their UI
+    // All deposit/withdraws done from their UI are always done with one/many UTXOs from the same address, and change sent back to the same address
+    // We also do this EXCLUSIVELY for THORChain Txs. The rest of the app uses xpubs, so the initially deposited from address isn't guaranteed to be populated
+    // if users send other UTXO Txs in the meantime after depositing
+    // Additionally, we select their highest balance UTXO address as a first deposit, which isn't guaranteed to contain enough value
+    //
+    // For both re/deposit flows, we will possibly need a pre-Tx to populate their highest UTXO/previously deposited from address with enough value
+
+    const withdrawInput = await getWithdrawInput()
+    if (!withdrawInput) throw new Error('Error building send input')
+
+    let txId: string
+
+    // Try/catching and evaluating to something in the catch isn't a good pattern usually
+    // In our case, handleSend() catching means that after all our previous checks, building a Tx failed at coinselect time
+    // So we actually send reconciliate a reconciliate Tx, retry the original send within the same block
+    // and finally evaluate to either the original Tx or a falsy empty string
+    try {
+      // 1. Try to deposit from the originally deposited from / highest UTXO balance address
+      // If this is enough, no other Tx is needed
+      txId = await handleSend({
+        sendInput: withdrawInput,
+        wallet: walletState.wallet,
+      })
+    } catch (e) {
```

Comment:
Safety factor:
-`chainId` check
- ensure it's a coinselect error


This should be brought to the deposit multiTxSend as well

Reply:
TBD in a follow-up PR to bring it to both and not clutter this diff